sky/resources.py

```diff
@@ -1209,4 +1184,7 @@ def __setstate__(self, state):
         if version < 12:
             self._docker_login_config = None
 
+        if version < 13:
+            state['_ports'] = [str(port) for port in state['_ports']]
```

Comment:
For the clusters before `version < 12` there may not be `state['_ports']`.

Also, `state['_ports']` can be None?

Reply:
Make sense! Resolved. Thanks!